### PR TITLE
Text fields are either varchar(255) or text.

### PR DESCRIPTION
**Pull request by atstojanov**
Consistent nullable fields between graphql and database models.

### Describe the background of your pull request

What does your pull request do? Does it solve a bug (which one?), add a feature?

### Additional context

Add any other context

### Governance 

- [ ] Documentation is added
- [ ] Test cases are added or updated
- [ ] I've read the contributing document https://github.com/cherrytwist/.github/blob/master/CONTRIBUTING.md
- [ ] I've read and understand the Code of Conduct https://github.com/cherrytwist/.github/blob/master/CODE_OF_CONDUCT.md
- [ ] I understand that any contributions or suggestions I made may make it into the actual code. I've read the License 
     https://github.com/cherrytwist/Coordination/blob/master/LICENSE 
     and the contributor license agreement: tba

### CODE REVIEW

File: src/domain/project/project.interface.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ export interface IProject {
   textID: string;
   name: string;
   description?: string;
-  state: string;
+  state?: string;
```
Comment (techsmyth) [inline src/domain/project/project.interface.ts]: state should be a required field, certainly moving forward - so suggest not to make it optional now

File: src/domain/profile/profile.dto.ts [hunk 2]
```diff
@@ -1,16 +1,13 @@
-import { InputType, Field } from '@nestjs/graphql';
-import { MaxLength } from 'class-validator';
```
Comment (techsmyth) [inline src/domain/profile/profile.dto.ts:2]: am guessing we want to leave these then in given the discussion this morning?

File: src/domain/project/project.entity.ts [hunk 3]
```diff
@@ -43,7 +43,7 @@ export class Project extends BaseEntity implements IProject {
       'The maturity phase of the project i.e. new, being refined, committed, in-progress, closed etc',
   })
   @Column({ nullable: true })
-  state: string;
+  state?: string;
```
Comment (techsmyth) [inline src/domain/project/project.entity.ts]: ditto - keep mandatory

File: src/domain/user/user.dto.ts [hunk 4]
```diff
@@ -1,50 +1,51 @@
...
+  @MaxLength(255)
   lastName?: string;
 
   @Field({
     nullable: true,
     description: 'Email address is required for creating a new user',
   })
-  @MaxLength(120)
+  @MaxLength(255)
+  @IsEmail()
```
Comment (techsmyth) [inline src/domain/user/user.dto.ts:28]: nice!

File: src/domain/challenge/challenge.interface.ts [hunk 5]
```diff
@@ -8,7 +8,7 @@ export interface IChallenge {
   id: number;
   name: string;
   textID: string;
-  state: string;
+  state?: string | null;
```
Comment (techsmyth) [inline src/domain/challenge/challenge.interface.ts]: keep mandatory

File: src/domain/challenge/challenge.entity.ts [hunk 6]
```diff
@@ -82,7 +82,7 @@ export class Challenge extends BaseEntity implements IChallenge, IGroupable {
       'The maturity phase of the challenge i.e. new, being refined, ongoing etc',
   })
   @Column({ nullable: true })
-  state: string;
+  state?: string;
```
Comment (techsmyth) [inline src/domain/challenge/challenge.entity.ts]: keep mandatory

File: src/domain/user/user.dto.ts (hunk 4 above)
Comment (techsmyth) [reply src/domain/user/user.dto.ts:28]: is it possible to create our own for e.g. textID?

**Review by techsmyth — CHANGES_REQUESTED** [review]
Nice update!

The state field does need to be mandatory as this will be used for lifecycles.

Also, need to take care that business rules related to lengths of eg tagline, textID are present on the dto objects.

File: src/domain/profile/profile.dto.ts (hunk 2 above)
Comment (atstoyanov) [reply src/domain/profile/profile.dto.ts:2]: Only for the description. The avatar is url. The url by specification can be up to 2MB. Which is not very likely to happen.

Comment (atstoyanov) [reply src/domain/profile/profile.dto.ts:2]: But we can add constrain IsUri

File: src/domain/user/user.dto.ts (hunk 4 above)
Comment (atstoyanov) [reply src/domain/user/user.dto.ts:28]: it is done for #571

**Review by techsmyth — APPROVED** [review]

**Review by valentinyanakiev — APPROVED** [review]

**Pull request merged**